### PR TITLE
No longer use WithId.

### PR DESCRIPTION
**Pull request by Lasering**
Use the latest openstack-common-clients.

### CODE REVIEW

File: src/main/scala/pt/tecnico/dsi/openstack/designate/services/Recordsets.scala [hunk 1]
```diff
@@ -4,20 +4,19 @@ import cats.effect.Sync
...
 import org.http4s.{Header, Query, Uri}
-import pt.tecnico.dsi.openstack.common.models.WithId
 import pt.tecnico.dsi.openstack.common.services.CrudService
 import pt.tecnico.dsi.openstack.designate.models.Recordset
 
 final class Recordsets[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
   extends CrudService[F, Recordset, Recordset.Create, Recordset.Update](baseUri, "recordset", authToken, wrapped = false) {
 
-  def getByName(name: String, extraHeaders: Header*): F[WithId[Recordset]] =
+  def getByName(name: String, extraHeaders: Header*): F[Recordset] =
```
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/designate/services/Recordsets.scala:13]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5936904)

File: src/main/scala/pt/tecnico/dsi/openstack/designate/services/Zones.scala [hunk 2]
```diff
@@ -5,17 +5,16 @@ import cats.syntax.flatMap._
...
 import org.http4s.{Header, Query, Uri}
-import pt.tecnico.dsi.openstack.common.models.WithId
 import pt.tecnico.dsi.openstack.common.services.CrudService
 import pt.tecnico.dsi.openstack.designate.models._
 
 final class Zones[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
   extends CrudService[F, Zone, Zone.Create, Zone.Update](baseUri, "zone", authToken, wrapped = false) { self =>
 
-  def getByName(name: String, extraHeaders: Header*): F[WithId[Zone]] =
+  def getByName(name: String, extraHeaders: Header*): F[Zone] =
```
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/designate/services/Zones.scala:14]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5936904)

File: src/main/scala/pt/tecnico/dsi/openstack/designate/DesignateClient.scala [hunk 3]
```diff
@@ -4,15 +4,18 @@ import cats.effect.Sync
...
 import org.http4s.{EntityDecoder, Header, Uri, circe}
-import pt.tecnico.dsi.openstack.common.models.WithId
 import pt.tecnico.dsi.openstack.designate.models.{Limit, Recordset}
 import pt.tecnico.dsi.openstack.designate.services.{FloatingIPs, Quotas, Recordsets, Zones}
 
 class DesignateClient[F[_]: Sync](baseUri: Uri, authToken: Header)(implicit client: Client[F]) {
-  val uri: Uri = if (baseUri.path.endsWith("v2") || baseUri.path.endsWith("v2/")) baseUri else baseUri / "v2"
+  val uri: Uri = {
+    val lastSegment = baseUri.path.dropEndsWithSlash.segments.lastOption
+    if (lastSegment.contains(Path.Segment("v2"))) baseUri else baseUri / "v2"
```
Comment (afonsomatos) [inline src/main/scala/pt/tecnico/dsi/openstack/designate/DesignateClient.scala]: What about
```scala
val uri: Uri = if (baseUri.path.dropEndsWithSlash.toString.endsWith("v2.0")) baseUri else baseUri / "v2.0"
```

**Review by afonsomatos — APPROVED** [review]
Looks good :)

File: src/main/scala/pt/tecnico/dsi/openstack/designate/DesignateClient.scala (hunk 3 above)
Comment (Lasering) [reply src/main/scala/pt/tecnico/dsi/openstack/designate/DesignateClient.scala]: Awesome, much better.

File: src/main/scala/pt/tecnico/dsi/openstack/designate/services/Recordsets.scala (hunk 1 above)
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/designate/services/Recordsets.scala:13]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5936904)

File: src/main/scala/pt/tecnico/dsi/openstack/designate/services/Zones.scala (hunk 2 above)
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/designate/services/Zones.scala:14]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5936904)

**Pull request merged**